src/sftp.rs

```diff
@@ -400,11 +401,15 @@ impl Sftp {
 
     fn lock(&self) -> Result<LockedSftp, Error> {
         match self.inner.as_ref() {
-            Some(inner) => {
-                let sess = inner.sess.lock();
+            Some(sftp_inner_drop_wrapper) => {
+                let sftp_inner = sftp_inner_drop_wrapper
+                    .0
+                    .as_ref()
+                    .expect("Never unset until shutdown, in which case inner is also unset");
```

Comment:
Since this method already return a Result, can't we raise an Error here instead of panicing?

Reply:
On a second thought, this is really a unlikely scenario where a LOT of things went wrong, hence panicing looks to be the right thing to do.

Reply:
Yes, more than an unlikely scenario, it should not be possible for any user to make that panic trigger, and if it does we want to get a report here => this is what panic is for, as explained by https://doc.rust-lang.org/book/ch09-03-to-panic-or-not-to-panic.html#cases-in-which-you-have-more-information-than-the-compiler